librz/core/linux_heap_glibc.c

```diff
@@ -1649,9 +1892,38 @@ static int GH(cmd_dbg_map_heap_glibc)(RzCore *core, const char *input) {
 			(core);
 		}
 		break;
+	case 'd': // "dmhd"
+		if (GH(rz_resolve_main_arena)(core, &m_arena)) {
```

Comment:
I usually prefer an approach with
```
if (<error>) {
   break/return;
}
... regularly executed code ...
```
as it avoids nested code, making code easier to read IMO.